riscv-debug-spec.tex

```diff
@@ -619,6 +642,34 @@ \subsection{JTAG Connector}
 connector looks as it does in Table~\ref{tab:header}. The function of each pin
 is described in Table~\ref{tab:pinout}.
 
+%MAW: Why do we only have 'RESET', but no 'TRST(n)' signal?
+% See http://openocd.org/doc/html/Reset-Configuration.html
+% Both are useful.
+% One is explicitly ``test logic reset'', which would imply
+% resetting the JTAG state machine and all associated test logic, in this
+% case the Debug Transport Module, synchronizers, and perhaps even
+% the Debug Module itself. It basically does the same thing as getting
+% the JTAG state machine into TEST_LOGIC_RESET state, which is
+% why it is optional.
+%
+% The other is a convenience which allows the debugger to essentially
+% hit the system reset button, which is basically mutually exclusive from
+% what TRST resets.
+%
+% I think a common workflow would be
+% to assert the halt signal in Debug Module,
+% then tell the debugger to pull the 'RESET' signal.
+% The desired effect woudld be to be able to debug the
+% code that executes on external reset.
+%
+% Another common workflow would be to have been debugging for a while,
+% then you disconnect the debugger.
+% Another user connects, and has no idea what state the debug logic
+% is in. They could assert the TRSTn signal (or clock JTAG TAP into
+% the proper state) and then know that all the Debug Transport Module
+% (and Debug Module logic...?) is in a known state.
```

Comment:
Maybe we just need a control bit in the Debug Module to reset its registers and any state (DTM->DM reset signal would be OK too, but we don't really have a concept of signals like that in this spec). Writing 1 to it from any transport would reset the Debug Module. Then a common debugger use case may be to assert RESET, assert Debug Module Reset, deassert RESET, deassert Debug Module Reset, now everything is in known state.

Actually, this could have other uses.... it could be that the Debug Module is in an inactive (reset) by default on power-on, then only when a debugger writes to it to "activate" it is it active. Knowing that it is "active" would be useful. For example, you could debug across a sleep routine by preventing actual power gating when debug is active. This would allow you to keep the Debug Module on the same power region as the components it can debug.